Scalars for extension dtypes.

TomAugspurger opened the pull request.

mrocklin commented:
@TomAugspurger what's the status on this?  Are you still working here?  Is this ready for people to review?

TomAugspurger commented:
Fixed the error failing CI. I'll need to look a bit more closely again before saying whether this is ready. I don't recall where I was when I dropped it.

TomAugspurger commented:
I've merged master to give the CI another run. I think this is ready to go, if you have a chance to look this over @jrbourbeau.

jrbourbeau commented:
Thanks for the update @TomAugspurger, I'll make time tomorrow to review this

jrbourbeau commented on dask/dataframe/utils.py:
```diff
@@ -445,15 +442,33 @@ def _scalar_from_dtype(dtype):
+
+@make_scalar.register(pd.Timestamp)
+@make_scalar.register(pd.Timedelta)
+@make_scalar.register(pd.Period)
```
Does it make sense for `make_scalar` to handle `pd.Categorical` types? Just thinking about the extensions built into pandas

TomAugspurger replied:
Categorical doesn't have a single scalar type like Timestamp or Period. It's a container for any scalar type, so there's no single type to register.

jrbourbeau replied:
Okay, thanks for clarifying

TomAugspurger commented on dask/dataframe/utils.py:
```diff
@@ -16,12 +16,9 @@
-    from pandas.api.types import is_interval_dtype
-except ImportError:
-    is_interval_dtype = lambda dtype: False
+from pandas.api.types import is_interval_dtype
```
Yes, is_interval_dtype / Interval is included in pandas 0.21.

jrbourbeau commented:
Merging on green
